tests/cross_functional/scale/noobaa/test_delete_objects.py

```diff
@@ -190,3 +202,122 @@ def test_delete_objects(
         # stop the io running in the background
         event.set()
         io_thread.result()
+
+    @pytest.fixture(scope="session")
+    def create_bucket_verify_object_deletion(
+        self,
+        request,
+        reduce_expiration_interval_session,
+        change_lifecycle_schedule_min_session,
+        bucket_factory_session,
+        mcg_obj_session,
+    ):
+        buckets = list()
+        expirations = list()
+
+        def factory(exp):
+            """
+            Factory function to create the bucket
+
+            """
+
+            # Reduce the expiration interval and lifecycle schedule delay
+            log.info(
+                "Reducing the expiration interval and lifecycle schedule delay to 1 minute"
+            )
+            reduce_expiration_interval_session(interval=1)
+            change_lifecycle_schedule_min_session(interval=1)
+
+            expirations.append(exp)
+            bucket = bucket_factory_session(amount=1, interface="OC")[0]
+            buckets.append(bucket)
+            return bucket
+
+        def teardown():
+            """
+            Teardown function to verify the object deletion
+            in the bucket
+
+            """
+            for bucket, expiration in zip(buckets, expirations):
+                verify_objs_deleted_from_objmds(bucket.name, timeout=30000, sleep=60)
+                if expiration:
+                    sample_if_objects_expired(
+                        mcg_obj_session, bucket.name, timeout=36000, sleep=60
+                    )
+                log.info("Verified that all objects are deleted and marked deleted")
+
+        request.addfinalizer(teardown)
+        return factory
+
+    @scale_long_run
+    @jira("DFBUGS-1106")
+    @jira("DFBUGS-1116")
+    @pytest.mark.parametrize(
+        argnames=["expiration"],
+        argvalues=[
+            pytest.param(
+                True,
+                marks=[
+                    pytest.mark.jira("DFBUGS-1116"),
+                    pytest.mark.polarion_id("OCS-6097"),
+                ],
+            ),
+            pytest.param(
+                False,
+                marks=[
+                    pytest.mark.jira("DFBUGS-1106"),
+                    pytest.mark.polarion_id("OCS-6096"),
+                ],
+            ),
+        ],
+    )
+    def test_delete_objs_by_expiration_and_recursive_deletion(
+        self,
+        create_bucket_verify_object_deletion,
+        mcg_obj_session,
+        awscli_pod_session,
+        test_directory_setup,
+        reduce_expiration_interval,
+        change_lifecycle_schedule_min,
```

Comment:
I think we can omit these two - we're already using their session-scoped versions in `create_bucket_verify_object_deletion`. 

Alternatively we can omit them from `create_bucket_verify_object_deletion` and keep them here.

Reply:
good catch!